From 567802d6e34ca3bfe60366662c2d1bd30d963543 Mon Sep 17 00:00:00 2001
From: Neal_Chen <neal_chen@wistron.com>
Date: Tue, 6 Jun 2023 11:25:23 +0800
Subject: [PATCH 5/6] Only import SFP related module

Signed-off-by: Neal_Chen <neal_chen@wistron.com>
---
 sonic_platform_base/__init__.py | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/sonic_platform_base/__init__.py b/sonic_platform_base/__init__.py
index 3d7f64b..b684159 100644
--- a/sonic_platform_base/__init__.py
+++ b/sonic_platform_base/__init__.py
@@ -1,11 +1,4 @@
-from . import chassis_base 
-from . import component_base
+from . import sonic_py_common
 from . import device_base
-from . import fan_base
-from . import fan_drawer_base
-from . import module_base
 from . import platform_base
-from . import psu_base
 from . import sfp_base
-from . import thermal_base
-from . import watchdog_base
-- 
2.25.1

