From dcc3b08c577aeecefd24b6090368f3b5c1e624d1 Mon Sep 17 00:00:00 2001
From: Neal_Chen <neal_chen@wistron.com>
Date: Wed, 24 Aug 2022 12:31:19 +0800
Subject: [PATCH] Add SFPHWMON device type

---
 include/devices.hpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/devices.hpp b/include/devices.hpp
index db57996c..f7fc450e 100644
--- a/include/devices.hpp
+++ b/include/devices.hpp
@@ -191,6 +191,9 @@ const boost::container::flat_map<const char*, ExportTemplate, CmpStr>
          {"SBTSI",
           ExportTemplate("sbtsi $Address", "/sys/bus/i2c/devices/i2c-$Bus",
                          "new_device", "delete_device", true)},
+         {"SFPHWMON",
+          ExportTemplate("sfphwmon $Address", "/sys/bus/i2c/devices/i2c-$Bus",
+                         "new_device", "delete_device", true)},
          {"SIC450",
           ExportTemplate("sic450 $Address", "/sys/bus/i2c/devices/i2c-$Bus",
                          "new_device", "delete_device", true)},
--
2.25.1

