From 6f24940ea6f916113aca59b3daac25e43e9936d9 Mon Sep 17 00:00:00 2001
From: Neal_Chen <neal_chen@wistron.com>
Date: Fri, 18 Nov 2022 18:15:48 +0800
Subject: [PATCH] Add pmbus_psu device type

Signed-off-by: Neal_Chen <neal_chen@wistron.com>
---
 include/devices.hpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/devices.hpp b/include/devices.hpp
index e7ed882b..b010734f 100644
--- a/include/devices.hpp
+++ b/include/devices.hpp
@@ -203,6 +203,9 @@ const boost::container::flat_map<const char*, ExportTemplate, CmpStr>
          {"pmbus",
           ExportTemplate("pmbus $Address", "/sys/bus/i2c/devices/i2c-$Bus",
                          "new_device", "delete_device", true)},
+         {"pmbus_psu",
+          ExportTemplate("pmbus_psu $Address", "/sys/bus/i2c/devices/i2c-$Bus",
+                         "new_device", "delete_device", true)},
          {"PXE1610",
           ExportTemplate("pxe1610 $Address", "/sys/bus/i2c/devices/i2c-$Bus",
                          "new_device", "delete_device", true)},
-- 
2.25.1

