From a6990346dfd9252b995295e4f4470cecde0c4521 Mon Sep 17 00:00:00 2001
From: jarvis_wc_chen <jarvis_wc_chen@wistron.com>
Date: Tue, 11 Apr 2023 14:48:10 +0800
Subject: [PATCH] add-device-type-powr1220

---
 include/devices.hpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/devices.hpp b/include/devices.hpp
index b01267c..d5b43f4 100644
--- a/include/devices.hpp
+++ b/include/devices.hpp
@@ -206,6 +206,9 @@ const boost::container::flat_map<const char*, ExportTemplate, CmpStr>
          {"pmbus_psu",
           ExportTemplate("pmbus_psu $Address", "/sys/bus/i2c/devices/i2c-$Bus",
                          "new_device", "delete_device", true)},
+         {"POWR1220",
+          ExportTemplate("powr1220 $Address", "/sys/bus/i2c/devices/i2c-$Bus",
+                         "new_device", "delete_device", true)},
          {"PXE1610",
           ExportTemplate("pxe1610 $Address", "/sys/bus/i2c/devices/i2c-$Bus",
                          "new_device", "delete_device", true)},
-- 
2.25.1

