From 49fd24a1d6202742cc131c7219e2663f6ab1c835 Mon Sep 17 00:00:00 2001
From: Neal Chen <neal_chen@wistron.com>
Date: Tue, 24 May 2022 18:24:28 +0800
Subject: [PATCH 1/2] jtag: Add support for aspeed JTAG adapter driver

Signed-off-by: Neal Chen <neal_chen@wistron.com>
---
 drivers/Kconfig  | 2 ++
 drivers/Makefile | 1 +
 2 files changed, 3 insertions(+)

diff --git a/drivers/Kconfig b/drivers/Kconfig
index 8708bedc549d..279801e4b14e 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -238,4 +238,6 @@ source "drivers/counter/Kconfig"
 source "drivers/most/Kconfig"
 
 source "drivers/peci/Kconfig"
+
+source "drivers/jtag/Kconfig"
 endmenu
diff --git a/drivers/Makefile b/drivers/Makefile
index 020780b6b4d2..8ae56ee1c290 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -188,3 +188,4 @@ obj-$(CONFIG_INTERCONNECT)	+= interconnect/
 obj-$(CONFIG_COUNTER)		+= counter/
 obj-$(CONFIG_MOST)		+= most/
 obj-$(CONFIG_PECI)		+= peci/
+obj-$(CONFIG_JTAG)		+= jtag/
-- 
2.25.1

