From 024f209eab4aa828c7bbe20b1c57d40aef9e6c5a Mon Sep 17 00:00:00 2001
From: Neal_Chen <neal_chen@wistron.com>
Date: Wed, 26 Apr 2023 14:00:01 +0800
Subject: [PATCH] drivers: mtd: spi-nor: gigadevice: Add gd55b01ge support

Signed-off-by: Neal_Chen <neal_chen@wistron.com>
---
 drivers/mtd/spi-nor/gigadevice.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mtd/spi-nor/gigadevice.c b/drivers/mtd/spi-nor/gigadevice.c
index 447d84bb2128..6c844e97b5c9 100644
--- a/drivers/mtd/spi-nor/gigadevice.c
+++ b/drivers/mtd/spi-nor/gigadevice.c
@@ -50,6 +50,9 @@ static const struct flash_info gigadevice_parts[] = {
 			   SPI_NOR_4B_OPCODES | SPI_NOR_HAS_LOCK |
 			   SPI_NOR_HAS_TB | SPI_NOR_TB_SR_BIT6)
 		.fixups = &gd25q256_fixups },
+	{"gd55b01ge", INFO(0xc8471B, 0, 64 * 1024, 2048,
+				SECT_4K | SPI_NOR_QUAD_READ | SPI_NOR_HAS_LOCK |
+				SPI_NOR_4B_OPCODES)},
 };
 
 const struct spi_nor_manufacturer spi_nor_gigadevice = {
-- 
2.25.1

